include/mgp_mock.py

```diff
@@ -0,0 +1,1011 @@
+
+    def __deepcopy__(self, memo):
+        # This is the same as the shallow copy, because we want to share the
+        # underlying C struct. Besides, it doesn't make much sense to actually
+        # copy _mgp_mock.Graph as that always references the whole graph state.
+        return Vertices(self._graph)
+
+    def is_valid(self) -> bool:
+        return True
+
+    def __iter__(self) -> typing.Iterable[Vertex]:
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        for id in self._graph.nodes:
+            yield Vertex(_mgp_mock.Vertex(id, self._graph))
+
+    def __contains__(self, vertex):
+        return self._graph.has_node(vertex.id)
+
+    def __len__(self):
+        if not self._len:
+            self._len = sum(1 for _ in self)
+
+        return self._len
+
+
+class Record:
+    __slots__ = ("fields",)
+
+    def __init__(self, **kwargs):
+        self.fields = kwargs
+
+
+class Graph:
+    __slots__ = ("_graph",)
+
+    def __init__(self, graph):
+        if not isinstance(graph, nx.MultiDiGraph):
+            raise TypeError(f"Expected 'networkx.classes.multidigraph.MultiDiGraph', got '{type(graph)}'")
+
+        self._graph = _mgp_mock.Graph(graph)
+
+    def __deepcopy__(self, memo):
+        # This is the same as the shallow copy, because we want to share the
+        # underlying C struct. Besides, it doesn't make much sense to actually
+        # copy _mgp_mock.Graph as that always references the whole graph state.
+        return Graph(self._graph)
+
+    def is_valid(self) -> bool:
+        return True
+
+    def get_vertex_by_id(self, vertex_id: VertexId) -> Vertex:
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        return Vertex(_mgp_mock.Vertex(vertex_id, self._graph.nx))
+
+    @property
+    def vertices(self) -> Vertices:
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        return Vertices(self._graph.nx)
+
+    def is_mutable(self) -> bool:
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        return not nx.is_frozen(self._graph.nx)
+
+    def create_vertex(self) -> Vertex:
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        if nx.is_frozen(self._graph.nx):
+            raise ImmutableObjectError("Cannot modify immutable object.")
+
+        new_id = max(node for node in self._graph.nx.nodes) + 1
+        self._graph.nx.add_node(new_id)
+
+        return Vertex(_mgp_mock.Vertex(new_id, self._graph.nx))
+
+    def delete_vertex(self, vertex: Vertex) -> None:
+        if not self.is_valid():
+            raise InvalidContextError()
+
+        if nx.is_frozen(self._graph.nx):
+            raise ImmutableObjectError("Cannot modify immutable object.")
+
+        if not nx.is_isolate(self._graph.nx, vertex.id):
+            raise LogicErrorError("Logic error.")
+
+        # TODO find out whether users can ever send this method a nonexistent vertex
```

Comment:
For when is this todo?

Reply:
I’ll remove this by the next review round